src/main/java/gov/cms/madie/terminology/service/InternalTerminologyService.java

```diff
@@ -0,0 +1,53 @@
+package gov.cms.madie.terminology.service;
+
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.server.exceptions.*;
+import gov.cms.madie.terminology.exceptions.HapiOperationException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.OperationOutcome;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.ValueSet;
+import org.springframework.stereotype.Service;
+
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class InternalTerminologyService {
+  private final IGenericClient hapiClient;
+
+  /**
+   * This method fetches the ValueSet expansion by id from HAPI server
+   *
+   * @param id -> Value Set id
+   * @return ValueSet -> Value Set with expansion details
+   */
+  public ValueSet getValueSetExpansionById(String id) {
+    try {
+      log.info("Fetching ValueSet expansion for {}", id);
+      Parameters parameters =
+          hapiClient
+              .operation()
+              .onInstance(new IdType("ValueSet", id))
+              .named("$expand")
+              .withNoParameters(Parameters.class)
+              .execute();
+      return (ValueSet) parameters.getParameter().get(0).getResource();
```

Comment:
Is there a chance of HAPI returning something other than a ValueSet Resource if it errors?

Reply:
because I'm specifying the instance on which the operation should be performed, I'll always get value set resource if it was successful operation. For errors it only throws reception. we don't get the resource in that case.